app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -20,13 +25,35 @@ open class FeedViewModel : ViewModel() {
     fun getNetworkErrorEvent(): LiveData<Event<String>> = networkErrorEvent
 
     init {
+        feedItems.addSource(repository.getFeedItems()) { items -> updateFeedItems(items) }
         refresh()
     }
 
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
+        isLoading.value = true
         isEmpty.value = true
+        loadFeedItems()
+    }
+
+    @SuppressLint("CheckResult")
+    fun loadFeedItems() {
+        repository.refresh()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ handleSuccess() }, { error -> handleNetworkError(error) })
```

Comment:
A good practice is to save the subscription in a `CompositeDisposable` and clear it when the `ViewModel` is cleared. 
This means, that the subscription will be disposed automatically when the user navigated aways (and the result won't be needed anymore).